editor/src/app/inputs/text-input.service.ts

```diff
@@ -44,16 +44,56 @@ export class TextInputService {
     if (!event.target.value && !this.hideIcon) {
       this.showIcon.next(true);
     }
-    this.updateField(event);
+    this.updateField(event, null);
   }
 
-  updateField(event) {
+  updateField(event, label) {
+
     if (event instanceof KeyboardEvent && (event.key === 'Escape' || event.keyCode === 27)) {
       (event.target as HTMLInputElement).value = this.lastValue;
       (event.target as HTMLInputElement).blur();
       return null;
     }
 
+    if (event.key === 'ArrowDown' || event.keyCode === 40 || event.key === 'ArrowUp' || event.keyCode === 38) {
+      const myLabel = ['Page margins', 'Margins', 'Gallery margins', 'Entry margins'];
+      const bigUnits = ['px', '%', 'pt'];
+      const litleUnits = ['em', 'rem', 'vw', 'vh'];
+
+      if (!myLabel.includes(label)) {
+        const regex  = /[a-z%]+/g;
+        const regex1 = /-?[0-9]\d*(\.\d+)?/g;
```

Comment:
why are your regexes called `regex` and `regex1`? what are they for?

Reply:
one regexe to get the unit regex;  and one for the negative and decimal number regex1
i have to rename it i think